src/compiler/crystal/semantic/main_visitor.cr

```diff
@@ -361,7 +361,11 @@ module Crystal
         end
 
         if meta_var.closured?
-          var.bind_to(meta_var)
+          unless var.bound_to?(meta_var)
```

Comment:
This code is already repeated three times, might be worth to extract it to a method.

Reply:
Done, thanks!